sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetUtils.scala

```diff
@@ -144,6 +144,42 @@ object ParquetUtils {
       file.getName == ParquetFileWriter.PARQUET_METADATA_FILE
   }
 
+  /**
+   * A StructField metadata key used to set the field id of a column in the Parquet schema.
+   */
+  val FIELD_ID_METADATA_KEY = "parquet.field.id"
```

Comment:
Did this config value come from Parquet-mr or is it something that you are proposing the PR?

Reply:
It is a spark metadata field we are proposing.

Reply:
I just want to make sure this does not clash with any existing ones in Spark or Parquet-mr.

Reply:
Yep, we didn't find anything existing in current spark or parquet. will note this in user facing change section.